### PR TITLE
Create DeleteMoneyRequest command

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
Removes `Cancel` and `Decline` money request functionality in favor of `Delete`

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/270502

### Tests
Pre-requisites: 2 accounts you have access to
Should be tested with the [Web-E](https://github.com/Expensify/Web-Expensify/pull/37076) and [Auth](https://github.com/Expensify/Auth/pull/7765) PRs

1. From account A, request money to account B `+ > Request money`
2. From account A, tap the request and verify that you see the `Delete` option
3. From account B, tap the request and verify that you don't see the `Delete` option
4. From account A, tap `Delete`
5. Verify that the request is deleted (the preview is not shown anymore) and you see a `deleted the $X request for comment` message in chat and LHN
6. Verify that the report total is updated to $0

- [ ] Verify that no errors appear in the JS console

### Offline tests
1. From account A, request money to account B `+ > Request money`
2. Go offline
3. From account A, tap `Delete`
4. Verify that the request is deleted (the preview is not shown anymore) and you see a `deleted the $X request for comment` message in chat and LHN, but the message is greyed out
6. Verify that the report total is updated to $0

### QA Steps
Same as steps above

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/22219519/233692774-c6d37dd0-0d5a-46d8-86b3-d433130a17e2.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/22219519/233692785-83be45ed-cc5a-4c1e-b65a-7f9a8acbe753.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/22219519/233692796-322bed6b-ec4a-4339-8aff-138ce25bfc67.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/22219519/233692803-6bdba639-98df-4edf-9929-8fd4d02a3a93.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/22219519/233692812-3e3660b5-19dc-4923-93a4-a2f239ca0357.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/22219519/233692824-2a78b44c-c844-442d-b9f7-8258a11e1a21.mov

</details>

### CODE REVIEW

File: src/libs/actions/IOU.js
Comment (mountiny) [inline src/libs/actions/IOU.js:551]: Should we drop the IOU terminology here since it can also be expense?
```suggestion
 * @param {Object} moneyRequestAction - the money request reportAction we are deleting
```

File: src/libs/actions/IOU.js
Comment (mountiny) [inline src/libs/actions/IOU.js:607]: Wondering if the transaction detail view should be pending too, ie if you could still load the page and it would be disabled essentially.

Thats probably too complicated though

File: src/pages/iou/IOUTransactions.js
Comment (mountiny) [inline src/pages/iou/IOUTransactions.js:85]:
```suggestion
                    const canBeDeleted = _.contains(deletableTransactions,
```

File: src/pages/iou/IOUTransactions.js
Comment (mountiny) [inline src/pages/iou/IOUTransactions.js:94]:
```suggestion
                            canBeDeleted={canBeDeleted}
```

**Review by mountiny — COMMENTED** [review]
Looking great, couple of comments only

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/pages/iou/IOUTransactions.js
Comment (mountiny) [reply src/pages/iou/IOUTransactions.js:60]: There still will be the cancel and declines from the past so we need to support it for displaying it, users wont be able to cancel or decline anymore

File: src/pages/iou/IOUTransactions.js
Comment (mountiny) [reply src/pages/iou/IOUTransactions.js:67]: even if you delete the transaction you will still have the `create` report action (of the one which was deleted) hanging around so _I think_ it should stay but havent looked into this deeply

File: src/libs/actions/IOU.js
Comment (mountiny) [reply src/libs/actions/IOU.js:609]: not sure if I follow the suggestion, you need to include the `optimisticReportAction` in the optimisticData as the deleted report action will be its own report action

Comment (s77rt) [reply src/libs/actions/IOU.js:609]: Oh my bad. You are right, that line should be kept even if it's not on it's own report action. Do the rest of the changes make sense?

Comment (mountiny) [reply src/libs/actions/IOU.js:609]: Hmm I am not sure. Just to be clear, we wont be removing the created and edited report actions, we will just add a new report action of type `delete` and what we will remove is the transaction.

Comment (s77rt) [reply src/libs/actions/IOU.js:609]: Yeah, I got that. We remove the transaction (delete it from onyx) and add one report action to act like history record for that transaction. I'm not suggesting changes that effect this area. I'm basically just questioning why we are no longer going to clear the `pendingAction` (other changes are just to make the code clearer)

**Comment by luacmartins** [comment]
Updated!

File: src/libs/actions/IOU.js
Comment (s77rt) [reply src/libs/actions/IOU.js:609]: @luacmartins Please address this

Comment (luacmartins) [reply src/libs/actions/IOU.js:609]: I posted a comment above - https://github.com/Expensify/App/pull/17583#discussion_r1175095560

Comment (s77rt) [reply src/libs/actions/IOU.js:609]: Oh sorry I had to refresh to see those :sweat_smile:

Personally I think that is a workaround. If the request failed then technically we have no pending action. I won't block on that though given that this seems to be a pattern now but we probably should decouple the pending action from errors.

**Comment by s77rt** [comment]
Should I hold testing till the two linked PRs are deployed?

**Comment by luacmartins** [comment]
Yes please

**Comment by s77rt** [comment]
@luacmartins Would you please tag me once those PRs are merged as I don't have access to any.

**Comment by luacmartins** [comment]
@s77rt will do.

File: src/libs/actions/IOU.js
Comment (mountiny) [reply src/libs/actions/IOU.js:607]: You ill have a transaction details page with $10 request for example and you will delete this transaction. Now the detailed page (`/r/:reportID/t/:transactionID`) can still be accessed as the thread, but how will we display it? the transaction data is null so we cannot show anything in the description or amount field.

I assume we will only have to show the original thread

File: src/libs/actions/IOU.js
Comment (mountiny) [inline src/libs/actions/IOU.js:658]: I think the comment is not needed here, its not used to lets remove it
```suggestion
```

**Review by mountiny — COMMENTED** [review]
Tested on the auth PR works fine on new requests but hitting some errors on older accounts and requests. Have you tested with those?

**Comment by luacmartins** [comment]
@mountiny I'm not sure that I understand that scenario. Is it something like:

1. Request money
2. Leave a comment on that request
3. Delete the request
4. Access the thread? 

If that's it, the thread would still show in the LHN, no? Also the route is `/r/:threadReportID` and at that point I don't think that we'd want the user to see the transaction data anymore and they'd only see the reportActions added to the thread, which would still contain things like `Changed the request from $X to $Y` since that's in the reportAction.

File: src/libs/actions/IOU.js
Comment (aldo-expensify) [inline src/libs/actions/IOU.js:577]: This is not related to the changes in this PR, but I strongly believe the code below should be corrected. Mutations are not good in a React project.

If you had not mutated `chatReport` below, these variables you are introducing here to keep the "original" values would not be needed.

**Review by aldo-expensify — DISMISSED** [review]
I think the error handling is not 100% correctly handled: I understand that the error message should only be displayed at the end of the red dot path, in a single place (see [this](https://github.com/Expensify/App/issues/15297#issuecomment-1470680706)). Here you show it in two places:

Screenshot 2023-04-28 at 5 25 36 PM
Screenshot 2023-04-28 at 5 25 41 PM

I see now that there seem to be some overlap with what is being fixed here: https://github.com/Expensify/App/issues/15297

File: src/libs/actions/IOU.js
Comment (s77rt) [reply src/libs/actions/IOU.js]: @luacmartins I think you don't need to clone the objects. We are using those values as read-only. I have suggested some similar change here https://github.com/Expensify/App/pull/17583#discussion_r1174327447 in code diff. 

_Please ignores the lines that remove `...optimisticReportAction` and add `pendingAction`._

**Comment by mountiny** [comment]
> If that's it, the thread would still show in the LHN, no? Also the route is /r/:threadReportID and at that point I don't think that we'd want the user to see the transaction data anymore and they'd only see the reportActions added to the thread, which would still contain things like Changed the request from $X to $Y since that's in the reportAction.

Ok I missed we changed this to not use the transactionID

**Review by mountiny — COMMENTED** [review]
https://user-images.githubusercontent.com/36083550/235272543-b21aa0d2-51d8-4617-b1f4-282899744bdf.mp4

Not sure what I am doing wrong, now I got transaction not found with the Web and Auth PRs all ready and build and auth restarted. I guess we will see with the C+ testing.

**Comment by aldo-expensify** [comment]
> Screen.Recording.2023-04-28.at.17.18.02.mp4 
> Not sure what I am doing wrong, now I got transaction not found with the Web and Auth PRs all ready and build and auth restarted. I guess we will see with the C+ testing.


I think there is something wrong with that video, it doesn't play :P

**Comment by luacmartins** [comment]
WRT [this comment](https://github.com/Expensify/App/pull/17583#pullrequestreview-1406265174), we're refactoring those views so we should see the error in the report view and won't see it in the transaction view anymore.

**Review by mountiny — DISMISSED** [review]
Looks good to me, hopefully we will deploy soon so we can complete the checklist

**Review by Julesssss — DISMISSED** [review]

**Comment by luacmartins** [comment]
Auth is deployed and Web-E is merged. We are still waiting on Web-E to make it to staging/prod before we can test/merge this though.

**Comment by mountiny** [comment]
@s77rt The API should be deployed to staging so can you please test against the staging API endpoint and complete the checklist? thank you very much 🙇

**Comment by s77rt** [comment]
@mountiny Can you please double chekc that both PRs deployed to staging? I'm not able to delete money request, getting 404 status code.

<details>

```js
 DEBUG  [alrt] ENSURE_BUGBOT unknown API request error caught while processing request
 {
    "message": "",
    "status": 404,
    "title": "",
    "request": {
        "command": "DeleteMoneyRequest",
        "data": {
            "transactionID": "4615189170863251301",
            "chatReportID": "6699736724742460",
            "reportActionID": "2784603409745294335",
            "iouReportID": "17710633238768",
            "appversion": "1.3.9-14",
            "apiRequestType": "write",
            "shouldRetry": true,
            "canCancel": true
        },
        "successData": [
            {
                "onyxMethod": "merge",
                "key": "reportActions_6699736724742460",
                "value": {
                    "2784603409745294335": {
                        "pendingAction": null
                    }
                }
            }
        ],
        "failureData": [
            {
                "onyxMethod": "merge",
                "key": "reportActions_6699736724742460",
                "value": {
                    "2784603409745294335": {
                        "errors": {
                            "1683125463677000": "Unexpected error deleting the money request, please try again later"
                        }
                    }
                }
            },
            {
                "onyxMethod": "merge",
                "key": "report_6699736724742460",
                "value": {
                    "lastMessageText": "Requested €10.00",
                    "lastMessageHtml": "requested €10.00",
                    "hasOutstandingIOU": true
                }
            },
            {
                "onyxMethod": "merge",
                "key": "report_17710633238768",
                "value": {
                    "hasOutstandingIOU": true,
                    "type": "iou",
                    "cachedTotal": "(DZD1,493.67)",
                    "chatReportID": "6699736724742460",
                    "currency": "DZD",
                    "managerEmail": "admin+test@abdelhafidh.com",
                    "ownerEmail": "admin@abdelhafidh.com",
                    "reportID": "17710633238768",
                    "state": "SUBMITTED",
                    "stateNum": 1,
                    "total": 149367,
                    "submitterPayPalMeAddress": "paypalz"
                }
            },
            {
                "onyxMethod": "set",
                "key": "transactions_4615189170863251301"
            }
        ]
    }
}
```

</details>


https://user-images.githubusercontent.com/16493223/235957758-94dabf7a-e9bb-47e9-83ff-d5b3c0cb22cc.mp4

**Comment by mountiny** [comment]
This should be in staging for sure, can you confirm on web if the requests are hitting staging server for sure?

**Comment by s77rt** [comment]
Right, on Web it's working well. Android/iOS didn't maybe the staging switch is not working correctly? I will check again and proceed with full tests asap.

BTW, We have some conflicts cc @luacmartins 

https://user-images.githubusercontent.com/16493223/236007614-4ad71ca4-d8c3-4d9d-a655-853926c52fc7.mp4

**Comment by s77rt** [comment]
@luacmartins 

> Verify that the request is deleted (the preview is not shown anymore) and you see a deleted the $X request for comment message in chat and LHN, but the message is greyed out

This is failing, the greyed out deleted message shows up but the preview is also still shown, maybe this is related to currency difference? (I will upload the videos asap)

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [X] Android / Chrome
    - [ ] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



https://user-images.githubusercontent.com/16493223/236050834-115ced80-0f38-466a-b6ee-09bfd2bfa3f3.mp4


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://user-images.githubusercontent.com/16493223/236050818-ec45ff16-b0d1-420c-b304-8792ed209966.mp4


</details>

<details>
<summary>Mobile Web - Safari</summary>


https://user-images.githubusercontent.com/16493223/236050818-ec45ff16-b0d1-420c-b304-8792ed209966.mp4

</details>

<details>
<summary>Desktop</summary>



https://user-images.githubusercontent.com/16493223/236050801-cc156790-4a10-47a8-99ce-3f187a7e9bc6.mp4


</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by s77rt** [comment]
I was not able to test this reliably on Native. Sometimes it works, sometimes it does not. I think this is not related to this PR but to the staging switch, it seems buggy.


getting `jsonCode: 400` response 

https://user-images.githubusercontent.com/16493223/236054201-6909d87a-6058-443a-97dd-8039f6bbb36c.mp4

**Comment by aldo-expensify** [comment]
Conflicts 😬

**Comment by luacmartins** [comment]
Conflicts resolved!

**Comment by luacmartins** [comment]
@s77rt it's possible that there's something weird going on with the staging switch. Hopefully we'll get these changed in production today and we can test it in prod soon.

**Comment by s77rt** [comment]
@luacmartins Please address the buggy preview https://github.com/Expensify/App/pull/17583#issuecomment-1533694841

**Comment by luacmartins** [comment]
@s77rt we are [changing](https://github.com/Expensify/App/pull/18143/files#diff-c8b4524175e6d85f4f031971dd6291346ddf18c49818d292fda1782128020bf5L73-L77) the behavior of the previews to always be shown, so they won't disappear anymore. I think it's fine to leave it as is

**Comment by s77rt** [comment]
@luacmartins Okay cool, so what's the next step here? We are basically done but I can't test on Native for the buggy staging switch. Should we wait till the PRs hit prod?

**Comment by luacmartins** [comment]
IMO we should just merge this given the urgency and address any issues as polish.

**Comment by s77rt** [comment]
I agree. The change is platform-independent anyway.

**Review by s77rt — APPROVED** [review]
LGTM! :rocket:

**Review by Julesssss — APPROVED** [review]
Approving latest changes.

File: src/libs/actions/IOU.js
Comment (s77rt) [reply src/libs/actions/IOU.js:625]: This is not a regression from this PR. We already added the report action in the optimistic data so we didn't need to add it in the failure data.

https://github.com/Expensify/App/blob/14ef88415c8fa93d659b5136ddb416be682acafa/src/libs/actions/IOU.js#L580-L590

**Pull request merged**
